tracee-ebpf/tracee/net_events.go

```diff
@@ -109,18 +228,32 @@ func (t *Tracee) processNetEvents() {
 					InterfaceIndex: idx,
 				}
 
-				err = t.pcapWriter.WritePacket(info, dataBuff.Bytes()[:pktLen])
+				_, pcapWriterExists := t.pcapWriters[packetContext]
+				if !pcapWriterExists {
+					err = t.createPcapFile(packetContext)
+					if err != nil {
+						t.handleError(err)
+						continue
+					}
+				}
+
+				err = t.pcapWriters[packetContext].WritePacket(info, dataBuff.Bytes()[:pktLen])
 				if err != nil {
 					t.handleError(err)
 					continue
 				}
 
 				// todo: maybe we should not flush every packet?
-				err = t.pcapWriter.Flush()
+				err = t.pcapWriters[packetContext].Flush()
 				if err != nil {
 					t.handleError(err)
 					continue
 				}
+			} else if (t.config.Output.PcapPerProcess && (netEventId == NetProcessExit || netEventId == NetContainerExit)) || (t.config.Output.PcapPerContainer && netEventId == NetContainerExit) {
+				_, pcapWriterExists := t.pcapWriters[packetContext]
+				if pcapWriterExists {
+					go t.netExit(packetContext)
+				}
```

Comment:
You can take care of the cleanup while processing sched_process_exit event instead of doing it here, and avoid the extra events sent in the network perf buffer

Reply:
ok. but then this map will be accessed from two different go routines. should i use a lock on this map?